### PR TITLE
add recompose v0.18.0 w/ npm autoupdate

### LINKED ISSUE #7620: [Request] Add recompose
**Library name:** recompose
**Git repository url:** https://github.com/acdlite/recompose
**npm package url(optional):** https://www.npmjs.com/package/recompose
**License(s):** MIT (according to [`package.json`](https://github.com/acdlite/recompose/blob/master/package.json))
**Official homepage:** https://github.com/acdlite/recompose
**Wanna say something? Leave message here:** I ❤️  React.js and functional programming, they works better if works together.

### PR DESCRIPTION
**Pull request by dannyxx001**
PR for #7620 
@pvnr0082t please help me review this PR, thanks.

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/acdlite/recompose
 * Official website (optional, not the repository): 
 * NPM package url (optional): https://www.npmjs.com/package/recompose
 * GitHub / Bitbucket popularity (required):
   - Count of stars: 57
   - Count of watchers: 1494
   - Count of forks: 67
 * NPM download stats (optional):
   - Downloads in the last day: 2202
   - Downloads in the last week: 10978
   - Downloads in the last month: 52515

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [x] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [x] More than 500 downloads stats per month on npm registry
 * [x] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [ ] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

close #7620, cc @acdlite

### CODE REVIEW

File: ajax/libs/recompose/package.json [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    "fbjs": "^0.8.1",
+    "hoist-non-react-statics": "^1.0.0",
+    "lodash": "^4.3.0"
+  },
+  "npmName": "recompose",
+  "npmFileMap": [
+    {
+      "basePath": "build",
+      "files": [
+        "Recompose*!(.json)"
```
Comment (pvnr0082t) [inline ajax/libs/recompose/package.json]: @dannyxx001 I think ```Recompose*.!(json)``` will be better.

**Comment by dannyxx001** [comment]
@pvnr0082t Sorry for late update. Please help me review again, thanks.

File: ajax/libs/recompose/package.json [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    "fbjs": "^0.8.1",
+    "hoist-non-react-statics": "^1.0.0",
+    "lodash": "^4.3.0"
+  },
+  "npmName": "recompose",
+  "npmFileMap": [
+    {
+      "basePath": "build",
+      "files": [
+        "Recompose*.!(json)"
```
Comment (PeterDaveHello) [inline ajax/libs/recompose/package.json]: Why not `"**/*.!(json"`?

Comment (pvnr0082t) [inline ajax/libs/recompose/package.json]: Oh! nice. I think `"**/*.!(json)"` is the best. :+1:

**Comment by dannyxx001** [comment]
OK, I have updated it. @pvnr0082t please help me review again, thanks.

**Comment by pvnr0082t** [comment]
@PeterDaveHello I think this PR is ok. Please review it again, thank you.

**Pull request merged**